wallet/src/plan.rs

```diff
@@ -1,14 +1,18 @@
+use rand_core::OsRng;
 use std::collections::{BTreeMap, HashMap};
 
 use anyhow::Result;
 use penumbra_component::stake::rate::RateData;
 use penumbra_component::stake::validator;
 use penumbra_crypto::{
-    asset::Denom, keys::AddressIndex, memo::MemoPlaintext, transaction::Fee, Address,
-    DelegationToken, FullViewingKey, Value, STAKING_TOKEN_ASSET_ID, STAKING_TOKEN_DENOM,
+    asset::Denom, dex::TradingPair, keys::AddressIndex, memo::MemoPlaintext, transaction::Fee,
+    Address, DelegationToken, FullViewingKey, Note, Value, STAKING_TOKEN_ASSET_ID,
+    STAKING_TOKEN_DENOM,
 };
 use penumbra_proto::view::NotesRequest;
-use penumbra_transaction::plan::{ActionPlan, OutputPlan, SpendPlan, TransactionPlan};
+use penumbra_transaction::plan::{
+    ActionPlan, OutputPlan, SpendPlan, SwapClaimPlan, SwapPlan, TransactionPlan,
```

Comment:
Nit: it would be good to try to resolve warnings at least before the next testnet (else we should do a pass e.g. like #1302 to avoid confusing testnet participants)